projects/ngrx.io/content/guide/signals/signal-store/entity-management.md

```diff
@@ -1,3 +1,397 @@
+```
+
+`withEntities` adds three properties of type `Signal` to the `TodoStore`.
+
+- `ids: Signal<EntityId[]>`: ids of all entities
+- `entities: Signal<Todo[]>`: array of all entities
+- `entityMap: Signal<EntityMap<Todo>>`> map of entities where the key is the id (`EntityId`)
+
+By default, `withEntities` requires your entity to have a property of name `id`, which serves as a unique identifier. `id` has to be of type `EntityId`, which is a `number` or a `string`.
+
+## Adding & Setting
+
+`addEntity` and `addEntities` add entries to the store. They act as updater functions for `patchState`.
+
+Here is an example on how to use them inside a component.
+
+```typescript
+@Component({
+  template: `@for (todo of todoStore.entities();track todo.id) {
+    <ul>
+      <li>{{ todo.name }}</li>
+    </ul>
+    }`,
+  standalone: true,
+  providers: [TodoStore],
+})
+export class EntityComponent implements OnInit {
+  todoStore = inject(TodoStore);
+
+  ngOnInit() {
+    // add a single entity
+    patchState(this.todoStore, addEntity({ id: 1, name: 'Car Washing', finished: false }));
+
+    // add multiple entities
+    patchState(
+      this.todoStore,
+      addEntities([
+        { id: 2, name: 'Car Washing', finished: false },
+        { id: 3, name: 'Room Cleaning', finished: false },
+      ])
+    );
+  }
+}
+```
+
+If you add an entity with an existing id, NgRx skips it silently, i.e., it doesn't throw an error.
+
+In this example, the todo will stick to "Cat Feeding".
+
+```typescript
+patchState(this.todoStore, addEntity({ id: 1, name: 'Cat Feeding', finished: false }));
+
+patchState(this.todoStore, addEntity({ id: 1, name: 'Dog Feeding', finished: false }));
+```
+
+---
+
+To override existing entities, use `setEntity` or `setEntities`.
+
+In this example, we add a new `Todo` entity with "Cat Feeding" and replace it with "Dog Feeding".
+
+```typescript
+patchState(this.todoStore, setEntity({ id: 1, name: 'Cat Feeding', finished: false }));
+
+patchState(this.todoStore, setEntity({ id: 1, name: 'Dog Feeding', finished: false }));
+```
+
+## Updating & Removing
+
+Updating and removing entities follow the same patterns as adding or setting them. The following functions are available:
+
+- Updating: `updateEntity`, `updateEntities`
+- Removing: `removeEntity`, `removeEntities`
+
+If an entity does not exist, it will not throw an error. This rule applies to all update and remove functions.
+
+The following example shows how to add, update and remove an entity.
+
+```typescript
+patchState(
+  this.todoStore,
+  addEntities([
+    { id: 2, name: 'Car Washing', finished: false },
+    { id: 3, name: 'Cat Feeding', finished: false },
+  ])
+);
+
+patchState(this.todoStore, updateEntity({ id: 2, changes: { finished: true } }));
+
+patchState(this.todoStore, removeEntity(3));
+```
+
+`updateEntity` requires as parameters the id to update and its changes (`Partial<Todo>`). `removeEntity` only requires the id.
```

Comment:
We should mention that `changes` can also be a function that returns [partial changes](https://github.com/ngrx/platform/blob/main/modules/signals/entities/src/models.ts#L32) and provide an example for this signature:

```ts
patchState(
  this.todoStore,
  updateEntity({ id: 2, changes: (todo) => ({ finished: !todo.finished }) })
);
```